### PR TITLE
feat:  #850 tab component

### PR DESCRIPTION
**Pull request by J0taFerreira**
- Implementation of primevue tab component
- Delegated admin tab is disabled until the respective table is available 
- The styling is on the common repo

### CODE REVIEW

**Comment by MCatherine1994** [comment]
Hi Jota, let's not put the tab component inside the "UserDataTable" component. It should be in the ManagePermission component, and include the UserDataTable component. Something like this:
````
<TabView>
         <TabPanel header="Users">
                   <UserDataTable />  // later on depends on which application is selected, we then decide to display UserDataTable or ApplicationAdminTable(new data table component will be created later)
         </TabPanel>
         <TabPanel header="Delegated admins">
                // here we will add the DelegatedAdminTable (new data table component will be created later)
         </TabPanel>
</TabView>
````
Screen Shot 2023-12-11 at 1 33 15 PM

File: frontend/src/components/managePermissions/ManagePermissions.vue [hunk 1]
```diff
@@ -34,8 +38,12 @@ const userRoleAssignments = shallowRef<
     FamApplicationUserRoleAssignmentGet[]
 >(props.userRoleAssignments);
 
+const passThroughTabStyles = {
```
Comment (MCatherine1994) [inline frontend/src/components/managePermissions/ManagePermissions.vue:41]: this variable doesn't seem in use?

File: frontend/src/components/managePermissions/ManagePermissions.vue [hunk 2]
```diff
@@ -81,69 +89,106 @@ async function deleteUserRoleAssignment(
...
+	.application-dropdown {
+		max-width: 38rem;
+	}
 
-    .dashboard-background-layout {
-        width: calc(100vw - 16rem) !important;
-    }
+	.dashboard-background-layout {
+		width: calc(100vw - 16rem) !important;
+	}
```
Comment (MCatherine1994) [inline frontend/src/components/managePermissions/ManagePermissions.vue:192]: Could you please check your format as well? Many changes in this file and the UserDataTable file are format change, which we shouldn't change them.

File: frontend/src/components/managePermissions/ManagePermissions.vue (hunk 1 above)
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissions.vue:41]: yeah, it was a left over, Thanks for pointing out

File: frontend/src/components/managePermissions/UserDataTable.vue [hunk 3]
```diff
@@ -16,351 +16,376 @@ import Button from '@/components/common/Button.vue';
...
-        <div v-if="!props.isApplicationSelected" class="no-app-selected">
-            <IconCapitol />
-            <p class="no-app-selected-title" no-app-selected>
-                Nothing to show yet!
-            </p>
-            <p class="no-app-selected-text">
-                Choose an application to show a list of users with access to it.
-                The list will display here.
-            </p>
-        </div>
```
Comment (MCatherine1994) [inline frontend/src/components/managePermissions/UserDataTable.vue:108]: Let's refactor line 99 to line 108 into a new component "TablePlaceholder.vue". And when there is no application selected, in ManagePermission.vue, show the TablePlaceholder. When the user select an application, we then display the tab and the UserDataTable. So you don't have to do style adjustment for tabs based on if application selected or not.

File: frontend/src/components/header/Header.vue [hunk 4]
```diff
@@ -40,6 +40,11 @@ const props = defineProps({
             </Button>
             <span class="header-title">
                 {{ props.title }}
+                <span class="env-label">
+                    <strong>
+                        {{ environmentLabel }}
+                    </strong>
+                </span>
```
Comment (MCatherine1994) [inline frontend/src/components/header/Header.vue:47]: just curious why we change the header?

Comment (J0taFerreira) [reply frontend/src/components/header/Header.vue:47]: The env label was not showing on mobile

File: frontend/src/components/managePermissions/UserDataTable.vue (hunk 3 above)
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/UserDataTable.vue:108]: I can do that

File: frontend/src/components/header/Header.vue (hunk 4 above)
Comment (MCatherine1994) [reply frontend/src/components/header/Header.vue:47]: um.... then we could make it more clean, like the following, so you don't need an extra class to hide the extra environmentLabel

```
 <span class="header-title">
        {{ props.title }}
         <strong class="subtitle">{{ props.subtitle }}</strong>
         <strong>{{ environmentLabel }}</strong>
 </span>
```

**Comment by J0taFerreira** [comment]
Closing this pr due to indenting size problem in formatter changing unnecessary files.
pr with correct files changed: https://github.com/bcgov/nr-forests-access-management/pull/1096

**Pull request closed without merging**